scripts/merge-changelogs.ts

```diff
@@ -0,0 +1,203 @@
+    /## (?<version>[0-9]+\.[0-9]+\..+)(?<content>[^]*?)(?=(\n## |$))/g;
+  const matches = changelog.matchAll(versionRegex);
+  const results: any[] = [];
+  for (const match of matches) {
+    results.push({
+      version: match.groups?.version,
+      content: match.groups?.content.trim()
+    });
+  }
+  return results;
+}
+
+function parseContent(
+  content: string,
+  version: string,
+  packageName: string
+): Message[] {
+  // I'm sorry... Explanation: https://regex101.com/r/9UhEwo/2
+  const contentRegex =
+    /- ((?<commit>.*):) (\[(?<type>.*?)\]) (?<message>[^]*?)(?=(\n- |\n### |$))|- Updated dependencies \[(?<depsCommit>.*)\](?<deps>[^]*?)\n*(?=(\n- |\n### |$))/g;
+  const results: Message[] = [];
+  for (const match of content.matchAll(contentRegex)) {
+    if (!match.groups?.commit && !match.groups?.depsCommit) {
+      throw new Error(
+        `No commit hash found for change set in ${packageName} for version ${version}!`
+      );
+    }
+    if (
+      !validMessageTypes.includes(match.groups?.type as any) &&
+      !match[0].includes('Updated dependencies')
+    ) {
+      throw new Error(
+        `Change set has an incorrect or missing type for change set in ${packageName} for version ${version}!`
+      );
+    }
+    const message = match.groups?.message?.trim() || match.groups?.deps?.trim();
+    if (message) {
+      results.push({
+        version,
+        message,
+        packageNames: [packageName],
+        commit: match.groups?.commit || match.groups?.depsCommit,
+        type: (match.groups?.type as any) || 'Updated Dependencies'
+      });
+    }
+  }
+  return results;
+}
+
+function parseChangelog(changelog: string): Message[] {
+  const packageName = getPackageName(changelog);
+  const versions = splitByVersion(changelog);
+  return versions
+    .map(({ version, content }) => parseContent(content, version, packageName))
+    .flat();
+}
+
+function writeHeader(version: string) {
+  return `
+# ${version}
+
+Release Date: TBD<br>
+API Docs: https://sap.github.io/cloud-sdk/api/${version}<br>
+Blog: TBD<br>`;
+}
+
+function writeMessagesOfType(
+  messages: Message[],
+  type: typeof validMessageTypes[number]
+): string {
+  const foo = messages.some(msg => msg.type === type)
+    ? `\n\n## ${type}\n\n` +
+      messages
+        .filter(msg => msg.type === type)
+        .map(
+          msg =>
+            `- [${msg.packageNames.join(', ')}] ${msg.message} (${msg.commit})`
+        )
+        .join('\n')
+    : '';
+  return foo;
+}
+
+function writeDependencyMessages(messages: Message[]) {
+  return messages.some(msg => msg.type === 'Updated Dependencies')
+    ? '\n\n## Updated Dependencies\n\n' +
+        messages
+          .filter(msg => msg.type === 'Updated Dependencies')
+          .map(
+            msg =>
+              `- [${msg.packageNames.join(', ')}] Updated Dependencies (${
+                msg.commit
+              })\n  ${msg.message}`
+          )
+          .join('\n')
+    : '';
+}
```

Comment:
What do you mean? Right now it would look like this:

```
## Updated Dependencies

- [connectivity, generator-common] Updated Dependencies (b57189c67)
  - @sap-cloud-sdk/util@2.4.0
- [generator] Updated Dependencies (b57189c67)
  - @sap-cloud-sdk/odata-v4@2.4.0
  - @sap-cloud-sdk/util@2.4.0
  - @sap-cloud-sdk/generator-common@2.4.0
  - @sap-cloud-sdk/odata-common@2.4.0
  - @sap-cloud-sdk/odata-v2@2.4.0
- [http-client, test-util] Updated Dependencies (b57189c67)
  - @sap-cloud-sdk/connectivity@2.4.0
  - @sap-cloud-sdk/util@2.4.0
- [odata-common, openapi] Updated Dependencies (b57189c67)
  - @sap-cloud-sdk/connectivity@2.4.0
  - @sap-cloud-sdk/util@2.4.0
  - @sap-cloud-sdk/http-client@2.4.0
- [odata-v2, odata-v4] Updated Dependencies (b57189c67)
  - @sap-cloud-sdk/connectivity@2.4.0
  - @sap-cloud-sdk/util@2.4.0
  - @sap-cloud-sdk/http-client@2.4.0
  - @sap-cloud-sdk/odata-common@2.4.0
- [openapi-generator] Updated Dependencies (b57189c67)
  - @sap-cloud-sdk/util@2.4.0
  - @sap-cloud-sdk/generator-common@2.4.0
  - @sap-cloud-sdk/openapi@2.4.0
```

Should certain packages not be considered in the `[]` or in the nested list? I also thought about removing this section completely from the global changelog, but I found it slightly interesting so I left it. Would removing completely be better?

Reply:
I hijacked your example and added `some-other-dependency` to it.
I think the information in you original example can be omitted completely, because there is no news, it just tells me that these packages depend on other packages.
For other packages this is more interesting, if we have this information cool. If not, I can live with that.

Expected result:
```
## Updated Dependencies

- [connectivity, generator-common] Updated Dependencies (b57189c67)
  - some-other-depency@1.1.0 
 ```

Reply:
It does not do that (AFAIK), so I will remove that section